### DISCUSSION TITLE
Copy the temp AE file with the changes

### DISCUSSION THREAD

**Discussion by Yehuda-V** [inlife/nexrender#956] [Q&A] [answered]
Hello,

I would like to know if there is a built-in option to copy the after-effect file created for the render with all the changes (for example, the replaced images or video, etc.) to a specific location, like the option to copy the final render to a specific location.

When I use this code in **myJob.json** file it copies the ae file without the changes:
   
```
    "actions":{
        "postrender": [
            {
                "module": "@nexrender/action-encode",
                "preset": "mp4",
                "output": "encoded.mp4"
            },
            {
                "module": "@nexrender/action-copy",
                "input": "encoded.mp4",
                "output": "myPath/encoded.mp4"
            },
            {
                "module": "@nexrender/action-copy",
                "input": "myproject.aep",
                "output": "myPath/myproject.aep"
            }
        ]
        
    }
```

**Comment by inlife** [comment, marked as answer]
Yeah, you did it all perfectly correctly. One minor thing is that nexrender does not save the project after the render is done, so you would need to call a js script that would do that.

Make sure to create a script.jsx with this content:
```js
app.project.save();
```

and then provide that file as an asset with "script" type

**Reply by Yehuda-V** [reply]
Thank you so much.

**Comment by Yehuda-V** [comment]
But the script will run before the render begins, there is a way to save the AE after the render completes?

**Reply by inlife** [reply]
Assets will be replaced before the render. It does not really matter if you save the project before or after, it will be the same.

**Reply by Yehuda-V** [reply]
Thanks

**Comment by Yehuda-V** [comment]
I now have a weird bug.
After running the code you suggested, `app.project.save();` I tried to open After Effects and the interface of the software looks strange and it is not possible to click and change any setting in the software.

Any idea what went wrong?
![image](https://github.com/inlife/nexrender/assets/66674483/bc667044-a1b5-48dc-b8ae-2bb61f6f2327)

**Reply by inlife** [reply]
It seems like After Effects was launched in render-only mode.

Please ensure that if you are using nexrender-worker you provide a flag to not add license. (worker adds this license by default)

Also, make sure to remove `ae_render_only_node.txt` file from My Documents folder and any Adobe related folders inside

**Reply by inlife** [reply]
For more information please refer to this guide: https://helpx.adobe.com/after-effects/kb/accidental-render-node-launch-ae.html

**Reply by Yehuda-V** [reply]
Thanks it works

**Comment by Yehuda-V** [comment]
It did indeed save the ae file but the images/videos files were missing.

My method to solve this is to tell the script directly to save the file in the desired location:

static method 
**Script.jsx**
`...app.project.save(File("g:/my/path"));...`

or 
dynamic method:
**myjob.json**
  ```
  "assets": [
        {
            "src": "file:///<replace this with you path>",
            "type": "script",
            "parameters": [
                {
                    "type": "string",
                    "key": "myPath",
                    "value": "<replace this with you path>/aeFileName.aep"
                }
            ]
        }
```
**script.jsx**
```
    var curPath = File(NX.get("myPath"));
    app.project.save(curPath);
```

**Comment by Yehuda-V** [comment]
Another problem I had was that the old image file was not deleted from the after-effect file.

I've fixed it, would you like me to add it as a pull request?
@inlife

**Reply by inlife** [reply]
Hey @Yon-Kyle 

What part would you see as a fix? Did you happen to find a bug?

**Reply by Yehuda-V** [reply]
yes, I have found and fixed the bug.

The bug was in replacing image assets in jsx (core > src > assets > nexrender.jsx) without deleting the previous image file form ae.

Here is the updated function (line 20):
```
nexrender.replaceFootage = function (layer, filepath) {
    if (!layer) { return false; }

    var file = new File(filepath);

    if (!file.exists) {
        throw new Error("nexrender: Trying to create a file replacement for an unknown file: " + filepath);
    }

    var importOptions = new ImportOptions(file);
    //importOptions.importAs = ImportAsType.COMP; // you can do stuff like this at this point for PSDs
    var theImport = app.project.importFile(importOptions);

>     var oldFootage = layer.source || null;

    layer.replaceSource(theImport, true);

>     if (oldFootage) oldFootage.remove();


    return true;
};

```

**Reply by inlife** [reply]
Ah, nice catch. Makes sense! A PR would be great!
